modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1329,16 +1330,53 @@ public String toDefaultValueWithParam(String name, Schema schema) {
      **/
     @SuppressWarnings("static-method")
     public String getSchemaType(Schema schema) {
-        // TODO better logic to handle compose schema
         if (schema instanceof ComposedSchema) { // composed schema
             ComposedSchema cs = (ComposedSchema) schema;
+            List<Schema> schemas = ModelUtils.getInterfaces(cs);
             if (cs.getAllOf() != null) {
                 for (Schema s : cs.getAllOf()) {
                     if (s != null) {
-                        // using the first schema defined in allOf
-                        schema = s;
-                        break;
+                        //schema = s;
                     }
+                    //LOGGER.info("ALL OF SCHEMA: {}", s);
+                }
+
+                LOGGER.info("Composed schema not yet supported: {}", cs);
+                // get the model (allOf)
+                return getAlias("UNKNOWN_COMPOSED_SCHMEA");
+            } else if (cs.getAnyOf() != null) { // anyOf
+                List<String> names = new ArrayList<String>();
+                for (Schema s : schemas) {
+                    if (StringUtils.isNotBlank(s.get$ref())) { // reference to another definition/schema
+                        String schemaName = ModelUtils.getSimpleRef(s.get$ref());
+                        if (StringUtils.isNotEmpty(schemaName)) {
+                            names.add(getAlias(schemaName));
+                        } else {
+                            LOGGER.warn("Error obtaining the datatype from ref:" + schema.get$ref() + ". Default to 'object'");
+                            return "object";
+                        }
+                    } else {
+                        // primitive type or model
+                        names.add(getAlias(getPrimitiveType(s)));
+                    }
+                    return "anyOf<" + String.join(",", names) + ">";
+                }
+            } else if (cs.getOneOf() != null) { // oneOf
+                List<String> names = new ArrayList<String>();
+                for (Schema s : schemas) {
+                    if (StringUtils.isNotBlank(s.get$ref())) { // reference to another definition/schema
+                        String schemaName = ModelUtils.getSimpleRef(s.get$ref());
+                        if (StringUtils.isNotEmpty(schemaName)) {
+                            names.add(getAlias(schemaName));
+                        } else {
+                            LOGGER.warn("Error obtaining the datatype from ref:" + schema.get$ref() + ". Default to 'object'");
+                            return "object";
+                        }
+                    } else {
+                        // primitive type or model
+                        names.add(getAlias(getPrimitiveType(s)));
+                    }
+                    return "oneOf<" + String.join(",", names) + ">";
```

Comment:
I'm guessing non-java langs will have to override this

Reply:
I think one way is to override getSchemaType in the generator and then parse the string, e.g.
```
String result = super.getSchemaType(schema);
// parse the string here e.g. oneOf<Cat, Dog>
```